src/test/java/uk/gov/hmcts/reform/bailcaseapi/domain/handlers/presubmit/ChangeDirectionDueDateHandlerTest.java

```diff
@@ -0,0 +1,280 @@
+import uk.gov.hmcts.reform.bailcaseapi.domain.entities.ccd.CaseDetails;
+import uk.gov.hmcts.reform.bailcaseapi.domain.entities.ccd.Event;
+import uk.gov.hmcts.reform.bailcaseapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.bailcaseapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.bailcaseapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.bailcaseapi.domain.entities.ccd.field.IdValue;
+
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ExtendWith(MockitoExtension.class)
+@SuppressWarnings("unchecked")
+class ChangeDirectionDueDateHandlerTest {
+
+    @Mock
+    private DateProvider dateProvider;
+    @Mock
+    private Callback<BailCase> callback;
+    @Mock
+    private CaseDetails<BailCase> caseDetails;
+    @Mock
+    private BailCase bailCase;
+
+    @Captor
+    private ArgumentCaptor<List<IdValue<Direction>>> bailValueCaptor;
+    @Captor
+    private ArgumentCaptor<BailCaseFieldDefinition> bailExtractorCaptor;
+
+    private LocalDate dateSent = LocalDate.now();
+
+    private ChangeDirectionDueDateHandler changeDirectionDueDateHandler;
+
+    @BeforeEach
+    public void setUp() {
+        when(dateProvider.now()).thenReturn(dateSent);
+
+        changeDirectionDueDateHandler =
+            new ChangeDirectionDueDateHandler(dateProvider);
+    }
+
+    @Test
+    void should_add_previous_dates_to_changed_direction() {
+
+        List<IdValue<Direction>> existingDirections =
+            Arrays.asList(
+                new IdValue<>("1", new Direction(
+                    "explanation-1",
+                    "Applicant",
+                    "2020-12-01",
+                    "2019-12-01",
+                    "",
+                    "",
+                    Collections.emptyList()
+                )),
+                new IdValue<>("2", new Direction(
+                    "explanation-2",
+                    "Home Office",
+                    "2020-11-01",
+                    "2019-11-01",
+                    "",
+                    "",
+                    Collections.emptyList()
+                ))
+            );
+
+        when(callback.getCaseDetails()).thenReturn(caseDetails);
+        when(callback.getEvent()).thenReturn(Event.CHANGE_BAIL_DIRECTION_DUE_DATE);
+        when(caseDetails.getCaseData()).thenReturn(bailCase);
+        when(bailCase.read(DIRECTIONS)).thenReturn(Optional.of(existingDirections));
+
+        // Direction 2 is selected to be changed (first field of DynamicList)
+        DynamicList dynamicList = new DynamicList(new Value("Direction 2", "Direction 2"),
+                                                  List.of(
+                                                      new Value("Direction 2", "Direction 2"),
+                                                      new Value("Direction 1", "Direction 1")));
+
+
+        // "Direction 1" in UI is equivalent of Direction with IdValue "2" in backend
```

Comment:
Is this comment right? based on the message Marcello put in chat?
Direction 1 -> IdValue 4 -> index 0
Direction 2 -> IdValue 3 -> index 1

Reply:
If you have 4 directions yes:
```
Direction 1 -> IdValue 4 -> index 0
Direction 2 -> IdValue 3 -> index 1
Direction 3 -> IdValue 2 -> index 2
Direction 4 -> IdValue 1 -> index 3
```

In the test there are two only:
```
Direction 1 -> IdValue 2 -> index 0
Direction 2 -> IdValue 1 -> index 1
```

Basically the Directions are ordered in the direction from most recent to oldest. 
The direction number is from 1 onwards.
The index in that collection is from 0 onwards.
The Idvalue is incremental but from oldest to newest, so if we have 4 directions it'll be 4, if we have 6 directions it'll be 6.

If you have 6 directions:
```
Direction 1 -> IdValue 6 -> index 0
Direction 2 -> IdValue 5 -> index 1
Direction 3 -> IdValue 4 -> index 2
Direction 4 -> IdValue 3 -> index 3
Direction 5 -> IdValue 2 -> index 4
Direction 6 -> IdValue 1 -> index 5
```